### PR TITLE
feat: add diff view when transpiling query

### PR DESCRIPTION
**Pull request by czgu**
add a comparison view before user confirm

image

### CODE REVIEW

File: querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    transpilerName,
+    toEngine,
+    fromEngine,
+    onHide,
+    onTranspileConfirm,
+}) => {
+    const { data: transpiledQuery, isLoading } = useResource(
+        useCallback(
+            () =>
+                TemplatedQueryResource.transpileQuery(
```
Comment (jczhong84) [inline querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx:38]: previously it was called with `await TemplatedQueryResource.transpileQuery()` directly, what's the difference and benefit to use `useResource`

File: querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }, [fromQuery, toQuery]);
+
+    return (
+        <div className="QueryComparison">
+            <div className="mr8 flex1">
+                <div className="center-align">
+                    <AccentText weight="bold">Original</AccentText>
+                </div>
+                <Tag>{fromEngine.name}</Tag>
+
```
Comment (jczhong84) [inline querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx:127]: nit: remove the empty line here.

File: querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx (hunk 1 above)
Comment (czgu) [reply querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx:38]: useResource is just a hook to auto load the data

File: querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+const QueryComparison: React.FC<{
+    fromQuery: string;
+    toQuery: string;
+    fromEngine: IQueryEngine;
+    toEngine: IQueryEngine;
+}> = ({ fromQuery, toQuery, fromEngine, toEngine }) => {
+    const [addedRanges, removedRanges] = useMemo(() => {
+        const added: IHighlightRange[] = [];
+        const removed: IHighlightRange[] = [];
+        const diffObjects = diffWordsWithSpace(fromQuery, toQuery);
```
[the remark this answers is not in the document]
Comment (czgu) [reply querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx:92]: maybe codemirror 6, it is kind of a hassle to use MergeView right now

File: querybook/webapp/ui/CodeHighlight/CodeHighlightWithMark.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+interface IProps {
+    query: string;
+    highlightRanges?: IHighlightRange[];
+
+    maxEditorHeight?: string;
+    autoHeight?: boolean;
+}
+
+export const CodeHighlightWithMark: React.FC<IProps> = ({
```
[the remark this answers is not in the document]
Comment (czgu) [reply querybook/webapp/ui/CodeHighlight/CodeHighlightWithMark.tsx:51]: sure, storybook doesn't run right now tho https://github.com/storybookjs/storybook/issues/15922

File: querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (czgu) [reply querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx]: i tried diffLines, but there is too much detail missing

**Review by jczhong84 — APPROVED** [review]

**Pull request merged**